detect.go

```diff
@@ -2,15 +2,21 @@ package gogenerate
 
 import (
 	"os"
+
 	"github.com/paketo-buildpacks/packit"
 )
 
-func Detect() packit.DetectFunc {
+func Detect(parser ConfigurationParser) packit.DetectFunc {
 	return func(context packit.DetectContext) (packit.DetectResult, error) {
 		run := os.Getenv("BP_GO_GENERATE")
 		if run != "true" {
 			return packit.DetectResult{}, packit.Fail.WithMessage("BP_GO_GENERATE is empty")
 		}
+
+		if _, err := parser.Parse(); err != nil {
+			return packit.DetectResult{}, packit.Fail.WithMessage("failed to parse go generate configuration: %w", err)
+		}
+
```

Comment:
Is it necessary to parse the generate configuration in detect? I feel like only parsing it in build is sufficient and would lead to more prominent failures due to misconfiguration. I worry that failing here will make the misconfiguration less obvious to the end user rather than failing at the beginning of the build process.

Reply:
That's fair. My initial choice was because I wanted the a quick feedback loop if someone's input was malformed (they'll see that the flag failed to parse during detect if they build with verbose output), rather than go through the potentially expensive steps of vendoring dependencies, etc. before failing the build due to a typo.

However, I agree, the failure is more obvious during build.

Reply:
Yeah. I am reluctant to fail in a meaningful way during detection. Especially in a buildpack that is most likely going to be marked optional.